Submitted By:            Igor Živković
Date:                    2015-07-15
Initial Package Version: 2.21
Upstream Status:         Not Applicable
Origin:                  FHS patch from LFS
Description:             This patch removes references to /var/db directory and
                         replaces them with more suitable directories in /var
                         hierarchy - /var/cache/nscd for nscd and
                         /var/lib/nss_db for nss_db. It also puts all the
                         binaries into /bin.

--- glibc-2.21.orig/Makeconfig
+++ glibc-2.21/Makeconfig
@@ -229,12 +229,12 @@
 
 # Where to install administrative programs.
 ifndef rootsbindir
-rootsbindir = $(exec_prefix)/sbin
+rootsbindir = $(exec_prefix)/bin
 endif
 inst_rootsbindir = $(install_root)$(rootsbindir)
 
 ifndef sbindir
-sbindir = $(exec_prefix)/sbin
+sbindir = $(exec_prefix)/bin
 endif
 inst_sbindir = $(install_root)$(sbindir)
 
@@ -259,7 +259,7 @@
 
 # Directory for the database files and Makefile for nss_db.
 ifndef vardbdir
-vardbdir = $(localstatedir)/db
+vardbdir = $(localstatedir)/lib/nss_db
 endif
 inst_vardbdir = $(install_root)$(vardbdir)
 
--- glibc-2.21.orig/nscd/nscd.h
+++ glibc-2.21/nscd/nscd.h
@@ -112,11 +112,11 @@
 
 
 /* Paths of the file for the persistent storage.  */
-#define _PATH_NSCD_PASSWD_DB	"/var/db/nscd/passwd"
-#define _PATH_NSCD_GROUP_DB	"/var/db/nscd/group"
-#define _PATH_NSCD_HOSTS_DB	"/var/db/nscd/hosts"
-#define _PATH_NSCD_SERVICES_DB	"/var/db/nscd/services"
-#define _PATH_NSCD_NETGROUP_DB	"/var/db/nscd/netgroup"
+#define _PATH_NSCD_PASSWD_DB	"/var/cache/nscd/passwd"
+#define _PATH_NSCD_GROUP_DB	"/var/cache/nscd/group"
+#define _PATH_NSCD_HOSTS_DB	"/var/cache/nscd/hosts"
+#define _PATH_NSCD_SERVICES_DB	"/var/cache/nscd/services"
+#define _PATH_NSCD_NETGROUP_DB	"/var/cache/nscd/netgroup"
 
 /* Path used when not using persistent storage.  */
 #define _PATH_NSCD_XYZ_DB_TMP	"/var/run/nscd/dbXXXXXX"
--- glibc-2.21.orig/nss/db-Makefile
+++ glibc-2.21/nss/db-Makefile
@@ -22,7 +22,7 @@
 		       /etc/rpc /etc/services /etc/shadow /etc/gshadow \
 		       /etc/netgroup)
 
-VAR_DB = /var/db
+VAR_DB = /var/lib/nss_db
 
 AWK = awk
 MAKEDB = makedb --quiet
--- glibc-2.21.orig/sysdeps/generic/paths.h
+++ glibc-2.21/sysdeps/generic/paths.h
@@ -68,7 +68,7 @@
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
+#define	_PATH_VARDB	"/var/lib/nss_db/"
 #define	_PATH_VARRUN	"/var/run/"
 #define	_PATH_VARTMP	"/var/tmp/"
 
--- glibc-2.21.orig/sysdeps/unix/sysv/linux/paths.h
+++ glibc-2.21/sysdeps/unix/sysv/linux/paths.h
@@ -68,7 +68,7 @@
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
+#define	_PATH_VARDB	"/var/lib/nss_db/"
 #define	_PATH_VARRUN	"/var/run/"
 #define	_PATH_VARTMP	"/var/tmp/"
 
